src/Surfnet/StepupGateway/SamlStepupProviderBundle/Controller/SamlProxyController.php

```diff
@@ -243,11 +245,19 @@ private function getIssuer(StateHandler $stateHandler): Issuer
     {
         // This can either be a SFO or 'regular' SSO authentication. Both use a ResponseContext service of their own
         $responseContextServiceId = $stateHandler->getResponseContextServiceId();
+        if (!$responseContextServiceId) {
```

Comment:
There are relatively often problems with the `StateHandler` and  maybe you could opt to refactor it to be more robust to prevent bugs in the future.

Reply:
I'm very reluctant to update it here and now. But yes you are right. The state handlers often cause confusion. Your comment did cause me to update the `gateway-state-gssp-registration-flow.png` diagram. Which aids in showing developers how/when state is used and for what purpose.